### PR TITLE
Adds `WPS242` which forbids to have too many cases in match block

### LINKED ISSUE #3202: Complexity: find PM with too many `case` statements
We need to report `match` with too many `case` statements. I guess 7+ should be the default limit.

### PR DESCRIPTION
**Pull request by pojknamn**
# I have made things!

## Checklist

- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x] I have created at least one test case for the changes I have made
- [x] I have updated the documentation for the changes I have made
- [x] I have added my changes to the `CHANGELOG.md`

## Related issues

- Closes #3202

### CODE REVIEW

File: tests/test_checker/test_noqa.py [hunk 1]
```diff
@@ -93,6 +93,7 @@
         'WPS239': 1,
         'WPS240': 0,  # only triggers on 3.12+
         'WPS241': 1,
+        'WPS242': 0,
```
Comment (sobolevn) [inline tests/test_checker/test_noqa.py:96]: please, add at least one to `noqa.py`. we need this to:
- make sure that `noqa` works for the correct node
- the whole integration process worked

File: wemake_python_styleguide/options/defaults.py [hunk 2]
```diff
@@ -143,6 +143,9 @@
 #: Maximum number of subjects in a ``match`` statement.
 MAX_MATCH_SUBJECTS: Final = 7  # 7 +- 0, guessed
 
+#: Maximum number of subjects in match statement.
+MAX_MATCH_CASES: Final = 6  # guessed
```
Comment (sobolevn) [inline wemake_python_styleguide/options/defaults.py:147]:
```suggestion
MAX_MATCH_CASES: Final = 7  # guessed
```

I think that 6 is not enough :(

**Review by sobolevn — COMMENTED** [review]
Thank you! Just one comment from my side :)

File: tests/test_checker/test_noqa.py (hunk 1 above)
Comment (pojknamn) [reply tests/test_checker/test_noqa.py]: oh, i`ve missed that, sorry.
Fixed

File: wemake_python_styleguide/options/defaults.py (hunk 2 above)
Comment (pojknamn) [reply wemake_python_styleguide/options/defaults.py]: Raised to 7 and fixed corresponding tests

**Review by sobolevn — APPROVED** [review]
Thanks a lot!

**Pull request merged**